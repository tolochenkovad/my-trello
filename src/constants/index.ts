export const COLLECTIONS = {
  tasks: 'tasks',
};