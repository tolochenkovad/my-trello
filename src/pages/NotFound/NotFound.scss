.not-found {
  display: flex;
  flex-direction: column;
  align-items: center;
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  font-size: 24px;

  a {
    padding: 10px;
    border-radius: 5px;
    margin-top: 20px;
    background-color: #0062cc;
    color: #ffffff;
    &:hover {
      text-decoration: none;
      background-color: #083668;
    }
  }
}