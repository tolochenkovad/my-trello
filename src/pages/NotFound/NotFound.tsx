import React from 'react';
import { Link } from 'react-router-dom';

const NotFound: React.FC = () => (
  <div className="not-found">
    <div>Oops, Page Not Found :(</div>
    <Link to="/">Go to main page</Link>
  </div>
);

export default NotFound;
