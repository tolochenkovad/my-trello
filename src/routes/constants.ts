export const ROUTES = {
  MAIN: '/',
  LOGIN: '/login',
  ANALYTICS: '/analytics',
};