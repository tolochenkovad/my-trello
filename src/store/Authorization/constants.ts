export const LOGOUT = `APP/authorization/LOGOUT` as const;
