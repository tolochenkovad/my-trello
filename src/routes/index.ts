import App from '../App';
import Container from '../components/Container';
import Login from '../components/Login';
import NotFound from '../components/NotFound';
import { RouteConfig } from 'react-router-config';

export const routes: RouteConfig[] = [
  {
    component: App,
    routes: [
      {
        component: Container,
        path: '/',
        exact: true,
      },
      {
        component: Login,
        path: '/login',
      },
      {
        component: Login,
        path: '/signup',
      },
      {
        component: NotFound,
      },
    ],
  },
];
