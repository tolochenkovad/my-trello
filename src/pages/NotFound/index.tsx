import React from 'react';
import { Link } from 'react-router-dom';
import { ROUTES } from '../../routes/constants';
import classes from './NotFound.module.scss';

const NotFound: React.FC = () => (
  <div className={classes.notFound}>
    <div>Oops, Page Not Found :(</div>
    <Link to={ROUTES.MAIN}>Go to main page</Link>
  </div>
);

export default NotFound;
