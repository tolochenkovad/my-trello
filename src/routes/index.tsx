import React, { FC } from 'react';
import { Route, Switch } from 'react-router-dom';
import Main from '../pages/Main';
import Authorization from '../pages/Authorization';
import NotFound from '../pages/NotFound';
import AuthRoute from './AuthRoute';
import Analytics from '../pages/Analytics';
import { ROUTES } from './constants';

const Routes: FC = () => (
  <div className="main-container">
    <Switch>
      <AuthRoute exact path={ROUTES.MAIN} component={Main} />
      <Route path={ROUTES.LOGIN} component={Authorization} />
      <AuthRoute path={ROUTES.ANALYTICS} component={Analytics} />
      <Route component={NotFound} />
    </Switch>
  </div>
);

export default Routes;
