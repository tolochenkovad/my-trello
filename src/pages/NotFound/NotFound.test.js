import React from 'react';
import { shallow } from 'enzyme';
import NotFound from './NotFound';
import { ROUTES } from '../../routes/constants';

let wrapper;

beforeEach(() => {
  wrapper = shallow(<NotFound />);
});

describe('NotFound', () => {
  it('should render', () => {
    expect(wrapper).toHaveLength(1);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('children is exist', () => {
    it('container', () => {
      expect(wrapper.find('.notFound')).toHaveLength(1);
    });

    it('Droppable', () => {
      expect(wrapper.find('Link').props().to).toEqual(ROUTES.MAIN);
    });
  });
});