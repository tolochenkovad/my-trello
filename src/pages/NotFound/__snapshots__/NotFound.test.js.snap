// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound should render correctly 1`] = `
<div
  className="not-found"
>
  <div>
    Oops, Page Not Found :(
  </div>
  <Link
    to="/"
  >
    Go to main page
  </Link>
</div>
`;
